core/capabilities/triggers/logevent/trigger.go

```diff
@@ -67,6 +67,11 @@ func newLogEventTrigger(ctx context.Context,
 		return nil, nil, err
 	}
 
+	err = contractReader.Start(ctx)
```

Comment:
can you expand why this change was needed in this PR?

Reply:
Yes -- previously this was being done in the testharness which is shared by the logevent trigger and by the workflow syncer. This caused the integration tests to pass for both even though any passed in contract readers would actually not work (because they would never be started.)

The fix is to cause the test to fail by not starting the contract reader in the test -- that happens here: https://github.com/smartcontractkit/chainlink/pull/15567/files#diff-23e2c7f1307a0cfa9724f11c5b20ffab5082fee238bffb021f54f605499fa427R121

And then to fix both things using the tests by starting the contract reader explicitly.